lib/elixir/src/elixir_expand.erl

```diff
@@ -262,7 +262,7 @@ expand({for, Meta, [_ | _] = Args}, E) ->
         {Args, []}
     end,
 
-  validate_opts(Meta, for, [do, into], Block, E),
+  validate_opts(Meta, for, [do, into, uniq], Block, E),
```

Comment:
Maybe something was not clear but uniq should be `true` or `false` at compilation time. This means you should validate here that it is actually a boolean and in elixir_erl_for you can assume that it is either true or false if present. That should simplify some of your code as well. :)

Reply:
Yeah, this simplified a lot the whole thing